models/productDao.js

```diff
@@ -0,0 +1,34 @@
+const { appDataSource } = require("./data-source");
+
+const showProducts = async (categoryId, limit, offset) => {
+  try {
+    let query = `SELECT
+          products.id AS products_id,
+          products.name AS products_name,
+          products.price,
+          products.thumbnail,
+          products.category_id,
+          categories.name AS category_name
+        FROM products
+        JOIN categories ON products.category_id = categories.id
+        ORDER BY products.id DESC
+      `;
+    if (categoryId !== "0") {
+      query += ` WHERE products.category_id = ?`;
+    }
+    query += ` LIMIT ? OFFSET ?`;
+    if (categoryId !== "0") {
+      return await appDataSource.query(query, [categoryId, limit, offset]);
+    } else {
+      return await appDataSource.query(query, [limit, offset]);
+    }
```

Comment:
같은 조건에 대한 반복문이 여러번 나옵니다. 다음과 같이 수정하면 조금 더 깔끔한 코드가 나올 것 같습니다. 한 번 확인해 보시고 수정해 보세요.
```suggestion
    let params = [limit, offset]
    let query = `SELECT
          products.id AS products_id,
          products.name AS products_name,
          products.price,
          products.thumbnail,
          products.category_id,
          categories.name AS category_name
        FROM products
        JOIN categories ON products.category_id = categories.id
        ORDER BY products.id DESC
      `;
    if (categoryId !== "0") {
      query += ` WHERE products.category_id = ?`;
      param = [categoryId, ...params]
    }
    
    query += ` LIMIT ? OFFSET ?`;

    return await appDataSource.query(query, params);
```